### PR TITLE
New post: Sagas vs workflows

### PR DESCRIPTION
**Pull request by hilton**

### CODE REVIEW

File: _posts/2017-03-29-saga-workflow.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+layout: article
+---
+
+Comparing sagas to workflow helps to clarify the more subtle and hidden features of workflows
+that are often overlooked.
+
+## The saga failure compensation pattern
+
+The term _saga_ was first used in 1987 in [a research paper by Hector Garcaa-Molrna
+Kenneth Salem](http://www.cs.cornell.edu/andru/cs711/2002fa/reading/sagas.pdf).
```
Comment (TimKam) [inline _posts/2017-03-29-saga-workflow.md]: _and_ Kenneth Salem

File: _posts/2017-03-29-saga-workflow.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
+---
+title: "Sagas vs workflows"
+description: "Sagas are not well known but provide inspiration for using workflows."
+tags: saga, workflow, transactions
+author: Tom Baeyens
+layout: article
+---
+
+Comparing sagas to workflow helps to clarify the more subtle and hidden features of workflows
```
Comment (TimKam) [inline _posts/2017-03-29-saga-workflow.md]: to workflow**s**

File: _posts/2017-03-29-saga-workflow.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+that are often overlooked.
+
+## The saga failure compensation pattern
+
+The term _saga_ was first used in 1987 in [a research paper by Hector Garcaa-Molrna
+Kenneth Salem](http://www.cs.cornell.edu/andru/cs711/2002fa/reading/sagas.pdf).
+It’s introduced as an conceptual alternative for long lived or distributed transactions.
+A saga is a larger transaction that is split up into a set of smaller transactions.
+This reduces the time that locks have to be held on database resources and therefore can
+avoid bottlenecks.  Each of the smaller transactions can be undone by a compensating action.
```
Comment (TimKam) [inline _posts/2017-03-29-saga-workflow.md]: Remove one space after _bottlenecks._

Comment (TimKam) [inline _posts/2017-03-29-saga-workflow.md]: Actually, there are a lot of double spaces after full stops `.` in this text.

File: _posts/2017-03-29-saga-workflow.md [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+A _workflow_, also known as a _business process_, is similar in the sense that it also defines a
+larger task as a set of individual activities.  The cool part is that you most often can specify
+graphically the ordering in which the activities have to be performed with arrows or transitions.
+So the focus of a workflow is on execution.  A workflow specifies an execution flow.  An arrow
+or transition is used to indicate that activities have to be executed sequentially.  Special
+constructs like parallel gateways allow for parallel execution of activities.
+
+![Parallel gateway workflow execution](../2017/book-trip-parallel.png)
+
+When Sagas are mentioned in the context of CQRS, they refer to workflows or process managers from
```
Comment (TimKam) [inline _posts/2017-03-29-saga-workflow.md]: I don't really understand why we introduce _CQRS_ (neither do I know that _CQRS_ is). Can we provide more context here?

File: _posts/2017-03-29-saga-workflow.md [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+the implementation perspective.  See
+[_A Saga on Sagas_ by Microsoft](https://msdn.microsoft.com/en-us/library/jj591569.aspx) and
+[_Sagas_ by Edument](http://www.cqrs.nu/Faq/sagas).
+
+## Saga as inspiration for a workflow feature
+There is already a great feature in Signavio Workflow that aligns nicely with the concept of sagas:
```
Comment (TimKam) [inline _posts/2017-03-29-saga-workflow.md]: We shouldn't assume readers know what Signavio Workflow is.

File: _posts/2017-03-29-saga-workflow.md (hunk 2 above)
Comment (hilton) [inline _posts/2017-03-29-saga-workflow.md]: `of` is better than `to`, although now you mention it, this should be `… workflows’ more subtle and hidden features`

File: _posts/2017-03-29-saga-workflow.md (hunk 3 above)
Comment (hilton) [inline _posts/2017-03-29-saga-workflow.md]: @tombaeyens The layout convention in tech blog post Markdown is to start a new line for each sentence, which makes it easier to see sentence length, and makes diffs more readable. The newlines aren't significant in Markdown.

File: _posts/2017-03-29-saga-workflow.md (hunk 4 above)
Comment (hilton) [inline _posts/2017-03-29-saga-workflow.md]: @tombaeyens I guess you need to explicitly state that ‘saga’ has a _different_ meaning in the context of CQRS.

File: _posts/2017-03-29-saga-workflow.md (hunk 5 above)
Comment (hilton) [inline _posts/2017-03-29-saga-workflow.md]: … so link to https://www.signavio.com/products/workflow/

**Pull request merged**